### PR TITLE
Show on report the scanapi version used to generate it (#379)

### LINKED ISSUE #379: Show on report the scanapi version used to generate it
After report generated, I seen that scan api version is not shown on anywhere. I think that can be useful to know what scanapi version is running in production.

### PR DESCRIPTION
**Pull request by hebertjulio**
Closes #379

### CODE REVIEW

File: scanapi/reporter.py [hunk 1]
```diff
@@ -42,4 +44,5 @@ def _build_context(results):
             "project_name": settings.get("project_name", ""),
             "results": results,
             "session": session,
+            "scanapi_version": get_distribution("scanapi").version,
```
Comment (Pradhvan) [inline scanapi/reporter.py:47]: @hebertjulio hey 👋🏾 Can you update the docstring to the method mentioned [here](https://github.com/scanapi/scanapi/wiki/First-Pull-Request#7-make-your-changes) in the wiki. Let's slowly self document our code. 🙌🏾 

Something like 
```
    @staticmethod
    def _build_context(results):
        """Formating context dict of values required to render template.
        Args:
            results [type]: Fill me!
        Returns:
            [dict]: values required to render template.
        """
        return {
            "now": datetime.datetime.now().replace(microsecond=0),
            "project_name": settings.get("project_name", ""),
            "results": results,
            "session": session,
            "scanapi_version": get_distribution("scanapi").version,
        }
```

I am not sure what would be the type of results and what would be the appropriate short description for it. 

Rest of the PR LGTM. Thank you for the help. 😄

**Review by Pradhvan — CHANGES_REQUESTED** [review]

File: scanapi/reporter.py (hunk 1 above)
Comment (hebertjulio) [reply scanapi/reporter.py:47]: Hey @Pradhvan,

I think results is a list of request, that right @camilamaia ?

Comment (camilamaia) [reply scanapi/reporter.py:47]: @hebertjulio, results is a generator object! Please, take a look at this answer, where you can see a more detailed explanation:

https://github.com/scanapi/scanapi/pull/384#discussion_r644227563

File: tests/unit/test_reporter.py [hunk 2]
```diff
@@ -64,6 +66,7 @@ def context(self, mocked__session):
             "project_name": "",
             "results": fake_results,
             "session": mocked__session,
+            "scanapi_version": get_distribution("scanapi").version,
```
Comment (camilamaia) [inline tests/unit/test_reporter.py:69]:
```suggestion
            "scanapi_version": '2.3.0',
```

One minor suggestions: as we don't want to test any external function, we can change it for what we expect: a string with a version. For us is irrelevant if the method `get_distribution` is working fine (we expect external methods were already tested externally).

File: tests/unit/test_reporter.py [hunk 3]
```diff
@@ -1,6 +1,8 @@
 from freezegun.api import FakeDatetime
 from pytest import fixture, mark
 
+from pkg_resources import get_distribution
```
Comment (camilamaia) [inline tests/unit/test_reporter.py:4]:
```suggestion

```

File: scanapi/reporter.py [hunk 4]
```diff
@@ -17,7 +19,12 @@ def __init__(self, output_path=None, template=None):
         self.template = template
 
     def write(self, results):
-        """Part of the Reporter instance that is responsible for writing scanapi-report.html."""
+        """Part of the Reporter instance that is responsible for writing scanapi-report.html.
+        Args:
+            results [generator]: list of endpoint node.
```
Comment (camilamaia) [inline scanapi/reporter.py:24]:
```suggestion
            results [generator]: generator of dicts resulting of Request run().
```

File: scanapi/reporter.py [hunk 5]
```diff
@@ -34,12 +41,16 @@ def write(self, results):
 
     @staticmethod
     def _build_context(results):
-        """Private method of Reporter returns dict containing keys datetime,
-        project_name, results and session and their corresponding values.
+        """Build context dict of values required to render template.
+        Args:
+            results [generator]: list of endpoint node.
```
Comment (camilamaia) [inline scanapi/reporter.py:46]:
```suggestion
            results [generator]: generator of dicts resulting of Request run().
```

**Review by camilamaia — APPROVED** [review]

**Comment by camilamaia** [comment]
Just wait for @Pradhvan's approval, so I can merge it :)

**Review by Pradhvan — APPROVED** [review]

**Comment by Pradhvan** [comment]
@camilamaia LGTM ⚡ 

Thanks for the help @hebertjulio 🤗

**Comment by hebertjulio** [comment]
Let's working next issue! :tada: :smile:

**Pull request merged**